content/en/blog/2019/alb-ingress-gateway-iks/index.md

```diff
@@ -0,0 +1,394 @@
+---
+title: Securely direct traffic from IBM Cloud Kubernetes Service Ingress to Istio Ingress Gateway
+description: Configure the IBM Cloud Kubernetes Service Application Load Balancer to direct traffic to Istio Ingress gateway with mutual TLS.
+subtitle: Configure the IBM Cloud Kubernetes Service Application Load Balancer to direct traffic to Istio Ingress gateway with mutual TLS
+publishdate: 2019-11-07
+attribution: Vadim Eisenberg (IBM)
+keywords: [traffic-management,ingress,file-mount-credentials,iks]
+---
+
+In this blog post I show how to configure the [Ingress Application Load Balancer (ALB)](https://cloud.ibm.com/docs/containers?topic=containers-ingress-about)
+on [IBM Cloud Kubernetes Service (IKS)](https://www.ibm.com/cloud/kubernetes-service/) to direct traffic to the Istio
+ingress gateway, while securing the traffic between them using {{< gloss  >}}mutual TLS authentication{{< /gloss >}}.
+
+When you use IKS without Istio, you may control your ingress traffic using the provided
+ALB. This ingress-traffic routing is configured using a Kubernetes
+[Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) resource with
+[ALB-specific annotations](https://cloud.ibm.com/docs/containers?topic=containers-ingress_annotation). IKS provides a
+DNS domain name for each cluster. The domain name is in the following format: `<cluster_name>.<region>.containers.appdomain.cloud`, for example
+`mycluster.us-south.containers.appdomain.cloud`. In addition to registering a DNS domain, IKS provides a TLS certificate
+that matches the domain, and a private key for the certificate. IKS stores the certificates and the private key in a
+[Kubernetes secret](https://kubernetes.io/docs/concepts/configuration/secret/).
+
+When you start using Istio in your IKS cluster, the recommended way is to use the
+[Istio Ingress Gateway](/docs/tasks/traffic-management/ingress/ingress-control/) instead of using the
+[Kubernetes Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/). One of the main reasons to use
+Istio ingress gateway is the fact the ALB provided by IKS will not be able to
+communicate directly with the services inside the mesh when you enable Istio mutual TLS. During your transition to
+having only Istio ingress gateway as your main entry point, you can continue to use the traditional Ingress for
+non-Istio services while using the Istio ingress gateway for services that are part of the mesh.
+
+IKS provides a convenient way for clients to access Istio ingress gateway by letting you
+[register a DNS domain](https://cloud.ibm.com/docs/containers?topic=containers-loadbalancer_hostname) for the gateway's
```

Comment:
```suggestion
[register a new DNS subdomain](https://cloud.ibm.com/docs/containers?topic=containers-loadbalancer_hostname) for the Istio ingress gateway's NLB 
```

Reply:
IKS allows registering a new DNS subdomain for any load balancer/gateway (IKS calls such a load balancer/gateway "NLB"). So Istio ingress gateway is a kind of such "NLB", so NLB is redundant here and it is rather IKS-specific term anyway.

Reply:
Addressed by https://github.com/istio/istio.io/pull/5098/commits/5c88b59a3df44e4402a5c9fc546e056e8e5f8da7 and https://github.com/istio/istio.io/pull/5098/commits/858333f67cb58829beef92a8c923259903f4bdec . Left out NLB, the point of the sentence is that IKS allows you to register a new subdomain for the IP address of the Istio Ingress Gateway. It is just the general using of subdomains: you register a subdomain for an IP address, so you will be able to use that subdomain instead of typing the IP address.